tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in

```diff
@@ -182,6 +183,47 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
               | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
               | View.SYSTEM_UI_FLAG_FULLSCREEN
               | API_SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+
+
+      if (getPackageManager().hasSystemFeature("android.software.leanback"))
+      {
+        // Leanback
+
+        new Thread()
+        {
+          public void run()
+          {
+            try
+            {
+              sleep(10000);
+              mJsonRPC.updateLeanback(Main.this);
+            } catch (Exception e)
+            {
+              e.printStackTrace();
+            }
+          }
+        }.start();
```

Comment:
This is awful :)

On each activity resume you start a new thread that update leanback every 10 seconds.

Meaning you start kodi then move to other app 2 or 3 times and now have 4 threads doing the same thing.

Thread should be started in onCreate if needed with an handler, then you send messages to the handler to trigger a refresh when needed. (Not sure every 10 second is a good value).